sale_product_set_variant/wizard/product_set_add.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2017-2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError, ValidationError
+from odoo.addons import decimal_precision as dp
+
+
+class ProductSetAddLine(models.TransientModel):
+    _name = 'product.set.add.line'
+    _order = 'sequence'
+
+    wiz_id = fields.Many2one(
+        'product.set.add',
+    )
+    source_line_id = fields.Many2one(
+        'product.set.line',
+    )
+    product_variant_ids = fields.Many2many(
+        'product.product',
+        domain="""[
+            '&',
+            ('sale_ok', '=', True),
+            ('product_tmpl_id', '=', product_template_id),
+        ]""",
+        string='Product',
+        required=False,
+    )
+    quantity = fields.Float(
+        string='Quantity',
+        digits=dp.get_precision('Product Unit of Measure'),
+    )
+    product_set_id = fields.Many2one(
+        'product.set',
+        string='Set',
+        related='source_line_id.product_set_id',
+        readonly=True,
```

Comment:
@bizzappdev no need to `readonly` when our field is `related`

Reply:
IMO explicit is better than implicit. having this as readonly=True  it should not harm the flow.
still if you want we can remove it.

Reply:
> IMO explicit is better than implicit. having this as readonly=True it should not harm the flow.
> still if you want we can remove it.

Ok, Not blocking anywhere, keep as it is